### ISSUE TITLE
Unable to page through property values when many subobjects

### ISSUE THREAD

**Issue by thingles** [SemanticMediaWiki/SemanticMediaWiki#1326]
Wasn't sure how to title this. You can see this issue at this URL

https://wikiapiary.com/w/index.php?title=Property:Has_remote_error&from=BruceTambling.com#SMWResults

The "from" value works, but this query returns more than 25 results. When you do "(next 25)" since it starts at the top you just get the same page back and you can’t page any further.

**Comment by mwjames** [comment]
> The "from" value works, but this query returns more than 25 results. When you do "(next 25)" since it starts at the top you just get the same page back and you can’t page any further.

I have seen in the past, it is an old issue (pre 1.8) and the problem sparks from how the page name is set and used for the next iteration.

For example, the first 25 are starting with "BruceTambling.com" so the next starts with "BruceTambling.com" as well which makes it a loop in loop and the search request goes in circles for "from=BruceTambling.com#SMWResults".

MW's webrequest object cuts the # subobject marker from the url request so the name will always be "BruceTambling.com" and not "BruceTambling.com#Extension_MediaPreview".

**Comment by thingles** [comment]
Yeah, figured that was the issue. Would there be a hack-y workaround to use some special character to pass the "#' into the URL and keep MediaWiki from slicing it off. Maybe even just URL encode the #? Or send something silly like "BruceTambling.com^Extension_MediaPreview" and mangle it on either end?

**Comment by mwjames** [comment]
> pass the "#' into the URL and keep MediaWiki from slicing it off. Maybe even just URL encode the #? Or send something silly like "BruceTambling.com^Extension_MediaPreview" and mangle it on either end?

OK, this is some old code but what if you try to replace part in `SMWPageLister` with 

```
-				$last = \SMW\StoreFactory::getStore()->getWikiPageSortKey( $this->mDiWikiPages[$resultCount - 1] );
+				$last = $this->mDiWikiPages[$resultCount - 1];
```

The `getWikiPageSortKey` returns a text without a subobject and quick test shows we encode `#` which should allow for the webrequest to fetch the whole string.

**Comment by thingles** [comment]
Well, that is better but it still isn't right. I put that change in my Semantic MediaWiki code locally so you can see it if you go to the URL and page around.

It does indeed allow the "(next 25)" links to work and you can page through it. However, it skips now. I guess I would take it skipping some over just refusing to page forward.

Now when I hit (next 25) it does proceed but it doesn't page inside the one set of subobjects right. It shows 25 of them, and then next page skips to the next site. I did [a query][1] and there are 46 of the subobjects for that one site so the 21 other ones are not visible in the paging. :-\

[1]: https://wikiapiary.com/w/index.php?title=Special%3AAsk&q=%5B%5BHas+website%3A%3ABruceTambling.com%5D%5D+%5B%5BHas+remote+error%3A%3A%2B%5D%5D&po=%3FHas+remote+error&eq=yes&p%5Bformat%5D=broadtable&sort_num=&order_num=ASC&p%5Blimit%5D=&p%5Boffset%5D=&p%5Blink%5D=all&p%5Bsort%5D=&p%5Bheaders%5D=show&p%5Bmainlabel%5D=&p%5Bintro%5D=&p%5Boutro%5D=&p%5Bsearchlabel%5D=...+further+results&p%5Bdefault%5D=&p%5Bclass%5D=sortable+wikitable+smwtable&p%5Bsep%5D=&eq=yes

**Comment by mwjames** [comment]
> Well, that is better but it still isn't right. I put that change in my Semantic MediaWiki code locally so you can see it if you go to the URL and page around.

The suggestion was made on a quick look at the old code and now that we know it doesn't work correctly, it means this needs a proper solution and effort. This will have to go on the back burner unless someone else steps up and sends a PR.

**Comment by thingles** [comment]
@mwjames Thanks for taking a look at it either way!

**Comment by mwjames** [comment]
Fixed with #1543.
